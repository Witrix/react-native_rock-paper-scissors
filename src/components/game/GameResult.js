import React, { Component } from 'react';
import {
    View,
    FlatList,
    Text,
    StyleSheet,
    TouchableHighlight,
} from 'react-native';
import AppStyle             from '../../themes/styles';
import AppColor             from '../../themes/colors';
import MyLocalization       from "../../my-lib/MyLocalzation";
import MyRules              from '../../my-lib/MyRules';
import Element from "./Element";
import Spacer from "../general/Spacer";

export default class GameResult extends Component {

    constructor(props) {
        super(props);
        this.state = {
            victory: props.victory,
            myElement: props.myElement,
            enemyElement: props.enemyElement,
            player: props.player,
        };
    }

    render() {
        let text = MyLocalization.getString('equality');
        let styleText = style.textEquality;
        if (this.state.victory === 1) {
            styleText = style.textVictory;
            text = MyLocalization.getString('victory');
        }
        else if (this.state.victory === -1) {
            styleText = style.textDefeat;
            text = MyLocalization.getString('defeat');
        }
        return (
            <View style={style.container}>
                <Text style={styleText}>{text}</Text>
                <Spacer size={20}/>
                <View style={style.players}>
                    <View style={style.sidePlayer}>
                        <Text style={style.textPlayer}>{this.state.player === 'ai' ? MyLocalization.getString('youAiPlayed') : MyLocalization.getString('youPlayed')}</Text>
                        <Spacer size={20}/>
                        <Element type={this.state.myElement}/>
                    </View>
                    <View style={style.sidePlayer}>
                        <Text style={style.textPlayer}>{MyLocalization.getString('enemyPlayed')}</Text>
                        <Spacer size={20}/>
                        <Element type={this.state.enemyElement}/>
                    </View>
                </View>
            </View>
        );
    }
}

const style = StyleSheet.create({
    container: {
        justifyContent: 'center',
        alignItems: 'center'
    },
    textPlayer: {
      fontSize: 20,
    },
    players: {
        flexDirection: 'row'
    },
    sidePlayer: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
    },
    textEquality: {
        fontSize: 40,
        color: 'black',
    },
    textVictory: {
        fontSize: 40,
        color: AppColor.brand.primary,
    },
    textDefeat: {
        fontSize: 40,
        color: AppColor.brand.secondary,
    }
});
