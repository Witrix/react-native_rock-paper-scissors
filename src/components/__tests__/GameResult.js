import 'react-native';
import React                    from 'react';
import GameResult               from '../game/GameResult';
import renderer                 from 'react-test-renderer';

it('Render correctly', () => {
    const tree = renderer.create(
        <GameResult />
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: valid props', () => {
    const tree = renderer.create(
        <GameResult victory={1}
                    myElement={'paper'}
                    enemyElement={'rock'}
                    player={'player'}
                    mode={'classic'}/>
    );
    expect(tree).toMatchSnapshot();
});

it('Render correctly: invalid props', () => {
    const tree = renderer.create(
        <GameResult victory={'abc'}
                    myElement={-1}
                    enemyElement={-1}
                    player={-1}
                    mode={-1}/>
    );
    expect(tree).toMatchSnapshot();
});