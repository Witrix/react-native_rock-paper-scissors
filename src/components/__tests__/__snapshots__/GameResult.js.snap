// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "fontSize": 40,
      }
    }
  >
    Equality
  </Text>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        You played
      </Text>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={null}
          style={
            Object {
              "height": 70,
              "width": 70,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Enemy played
      </Text>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={null}
          style={
            Object {
              "height": 70,
              "width": 70,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Render correctly: invalid props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "black",
        "fontSize": 40,
      }
    }
  >
    Equality
  </Text>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        You played
      </Text>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={null}
          style={
            Object {
              "height": 70,
              "width": 70,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Enemy played
      </Text>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={null}
          style={
            Object {
              "height": 70,
              "width": 70,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`Render correctly: valid props 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#29B6F6",
        "fontSize": 40,
      }
    }
  >
    Victory
  </Text>
  <View
    style={
      Object {
        "marginTop": 20,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        You played
      </Text>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 70,
              "width": 70,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Paper
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        Enemy played
      </Text>
      <View
        style={
          Object {
            "marginTop": 20,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 70,
              "width": 70,
            }
          }
        />
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
        >
          Rock
        </Text>
      </View>
    </View>
  </View>
</View>
`;
